natscontext/context.go

```diff
@@ -99,6 +99,34 @@ func New(name string, load bool, opts ...Option) (*Context, error) {
 	return c, nil
 }
 
+// NewFromFile loads a new configuration context from the given filename.
+//
+// When opts is supplied those settings will override what was loaded or supply
+// values for an empty context
+func NewFromFile(filename string, opts ...Option) (*Context, error) {
+	c := &Context{
+		Name:   strings.TrimSuffix(filepath.Base(filename), filepath.Ext(filepath.Base(filename))),
+		config: &settings{},
+		path:   filename,
+	}
+
+	err := c.loadActiveContext()
```

Comment:
Looks good in general, would be nice to take the code from this line down and put it in a helper function so that it can be shared and reused between New and NewFromFile which would avoid mistakes made later should we forget to update one of these methods with new behaviors.

Otherwise all good

Reply:
Makes sense - added a helper